File: menu.py
from tabuleiro import Tabuleiro
from cavalo import Cavalo
from torre import Torre
from bispo import Bispo


class Menu:
    # Atrbutos estáticos da classe
    opcao = 0
    t = Tabuleiro()

    @classmethod
    def ler_opcao(cls, quant_opcoes):
        cls.opcao = int(input("Opção: "))
        while cls.opcao < 1 or cls.opcao > quant_opcoes:
            print(f"Opção Inválida! Selecione uma opção entre 0 e {quant_opcoes}.")
            cls.opcao = int(input("Opção: "))

    @staticmethod
    def exibir_menu():
        print("1 - Adicionar Bispo.")
        print("2 - Adicionar Cavalo.")
        print("3 - Adicionar Torre.")
        print("4 - Movimentar Peça.")
        print("5 - Exibir tabuleiro.")
        print("6 - Sair")

    @classmethod
    def adicionar_peca(cls):
        posicao_x = int(input("Informe a posição de origem x: ")) - 1
        posicao_y = int(input("Informe a posição de origem Y: ")) - 1
        cor = int(input("Informe a cor da peça (1 para 'Branco' ou 2 para 'Preto'): "))
        cor = cls.seleciona_cor(cor)

        if cls.opcao == 1:
            cls.t.pecas = Bispo(posicao_x, posicao_y, cor)
        elif cls.opcao == 2:
            cls.t.pecas = Cavalo(posicao_x, posicao_y, cor)
        else:
            cls.t.pecas = Torre(posicao_x, posicao_y, cor)

    @classmethod
    def seleciona_cor(cls, cor_numero):
        if cor_numero == 1:
            return "Branco"
        elif cor_numero == 2:
            return "Preto"

        return "Branco"

    @classmethod
    def recuperar_peca(cls, posicao_atual_x, posicao_atual_y):
        peca = cls.t.matriz_posicao[posicao_atual_x][posicao_atual_y]

        while peca == 0:
            print("Posição Inválida! Não existe peça na posição indicada", end=". ")
            print("Informe uma posição válida!")
            posicao_atual_x = int(input("Posição X: ")) - 1
            posicao_atual_y = int(input("Posição Y: ")) - 1
            peca = cls.t.matriz_posicao[posicao_atual_x][posicao_atual_y]
        return peca

    @classmethod
    def realizar_movimento(cls):
        print("Informe as coordenadas da peça que você deseja movimentar: ")
        atual_x = int(input("Posição X: ")) - 1
        atual_y = int(input("Posição Y: ")) - 1

        peca = cls.recuperar_peca(atual_x, atual_y)

        print("Mover a peça para: ")
        destino_x = int(input("Destino X: ")) - 1
        destino_y = int(input("Destino Y: ")) - 1

        cls.t.movimentar_peca(destino_x, destino_y, peca.id)


    @classmethod
    def exibir_tabuleiro(cls):
        print()
        cls.t.imprimir_tabuleiro()
        print()


# Rotina para o jogo
quantidade_opcoes = 6

while Menu.opcao != quantidade_opcoes:
    Menu.exibir_menu()
    Menu.ler_opcao(quantidade_opcoes)

    if 1 <= Menu.opcao <= 3:
        Menu.adicionar_peca()
        Menu.exibir_tabuleiro()
    elif Menu.opcao == 4:
        Menu.realizar_movimento()
        Menu.exibir_tabuleiro()
    elif Menu.opcao == 5:
        Menu.exibir_tabuleiro()
    else:
        print("Obrigada por jogar!")
        break

    input("Aperte qualquer tecla para continuar...")
    print()

File: src/xadrez_jv/Menu.java
package xadrez_jv;

import java.awt.*;
import javax.swing.*;

import xadrez_jv.Menu;

public class Menu extends JFrame{
	public static int opcao = 1;
	public static Tabuleiro tabuleiro = new Tabuleiro();
	//public static Scanner s = new Scanner(System.in);
	
	//Atributos da interface gráfica
	private ImageIcon icon;
	private JLabel lblimagem;
	private JButton btnexplicavel;
	private JButton btnsimulacao;
	private JPanel cLeftSide;
	private JPanel cRightSide;
	
	Menu()
	{
		icon = new ImageIcon(getClass().getResource("/imagens/chess.jpeg"));
		lblimagem = new JLabel(icon);
		btnexplicavel = new JButton("Modo Explicável");
		btnsimulacao = new JButton("Modo Simulação");
		
		/*btnexplicavel.setPreferredSize(new Dimension(100, 30));
		btnsimulacao.setPreferredSize(new Dimension(100, 30));
		btnexplicavel.setMargin(new Insets(5,5,5,5));
		btnsimulacao.setMargin(new Insets(5,5,5,5));*/
		
		cRightSide = new JPanel();
		cRightSide.setLayout(new GridLayout(1, 1));
		cRightSide.add(lblimagem);
		
		cLeftSide = new JPanel();
		cLeftSide.setLayout(new GridLayout(2, 1));
		cLeftSide.add(btnexplicavel);
		cLeftSide.add(btnsimulacao);
		
		setLayout(new GridLayout(1, 2));
		add(cLeftSide);
		add(cRightSide);
	}
	
	/*public static void lerOpcao(int quantidadeOpcoes)
	{
		System.out.print("Opção: ");
		Menu.opcao = s.nextInt();
		System.out.println();
		
		while (Menu.opcao < 1 || Menu.opcao > quantidadeOpcoes)
		{
			System.out.println("Opção Inválida! Selecione uma opção entre 0 e " + quantidadeOpcoes);
			Menu.opcao = s.nextInt();
			System.out.println();
		}
	}
	
	public static void exibirMenu()
	{
		System.out.println("1 - Adicionar Bispo");
		System.out.println("2 - Adicionar Cavalo");
		System.out.println("3 - Adicionar Torre");
		System.out.println("4 - Movimentar Peça");
		System.out.println("5 - Exibir Tabuleiro");
		System.out.println("6 - Sair");
	}
	
	public static void adicionarPeca()
	{
		System.out.print("Informe a posição X: ");
		int posicaoX = s.nextInt() - 1;
		System.out.println("X: " + posicaoX);
		System.out.println("Informe a posição Y: ");
		int posicaoY = s.nextInt() - 1;
		System.out.println("Y: " + posicaoY);
		System.out.println("Informe a cor (1 para 'Branco' e 2 para 'Preto'): ");
		int cor = s.nextInt() - 1;
		System.out.println();
		
		Peca peca;
		
		if (Menu.opcao == 1) {
			peca = new Bispo(posicaoX, posicaoY, cor);
			System.out.println("BISPO X: " + peca.posicaoAtualX + " BISPO Y: " + peca.posicaoAtualY);
		}else if(Menu.opcao == 2) 
			peca = new Cavalo(posicaoX, posicaoY, cor);
		else
			peca = new Torre(posicaoX, posicaoY, cor);
		
		Menu.tabuleiro.setPeca(peca);
	}
	
	public static Peca recuperarPeca(int posicaoAtualX, int posicaoAtualY)
	{
		Peca peca = Menu.tabuleiro.getMatrizPosicao(posicaoAtualX, posicaoAtualY);
		
		while (peca == null)
		{
			System.out.println("Posição Inválida! Não existe peça na posição indicada. Informe uma posição válida!");
			System.out.println("Posição X: ");
			posicaoAtualX = s.nextInt() - 1;
			System.out.println("Posição Y: ");
			posicaoAtualY = s.nextInt() - 1;
			peca = Menu.tabuleiro.getMatrizPosicao(posicaoAtualX, posicaoAtualY);
		}
		
		return peca;
	}
	
	public static void realizarMovimento()
	{
		System.out.println("Informe as coordenadas da peça que você deseja movimentar: ");
		System.out.print("Posição X: ");
		int atualX = s.nextInt() - 1;
		System.out.println();
		System.out.print("Posição Y: ");
		int atualY = s.nextInt() - 1;
		System.out.println();
		
		Peca peca = recuperarPeca(atualX, atualY);
		
		System.out.println("Mover a peça para: ");
		System.out.print("Destino X: ");
		int destinoX = s.nextInt() - 1;
		System.out.println();
		System.out.print("Destino Y: ");
		int destinoY = s.nextInt() - 1;
		System.out.println();
		
		Menu.tabuleiro.movimentarPeca(destinoX, destinoY, peca.id);
	}
	
	public static void exibirTabuleiro()
	{
		System.out.println();
		Menu.tabuleiro.imprimirTabuleiro();
		System.out.println();
	}*/
}
